packages/next-app/src/app/stableswapsimulator/(components)/TokenTable.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import {
+  MinusCircledIcon,
+  Pencil1Icon,
+  PlusCircledIcon,
+} from "@radix-ui/react-icons";
+
+import { Dialog } from "#/components/Dialog";
+import Table from "#/components/Table";
+import { TokensData, useStableSwap } from "#/contexts/StableSwapContext";
+
+import TokenForm from "./TokenForm";
+
+export function TokenTable() {
+  const { initialData } = useStableSwap();
+
+  return (
+    <div className="h-full flex-1 flex w-full justify-center text-white">
+      <Table>
+        <Table.HeaderRow>
+          <Table.HeaderCell>
+            <span className="sr-only">Edit</span>
+          </Table.HeaderCell>
+          <Table.HeaderCell>Symbol</Table.HeaderCell>
+          <Table.HeaderCell>Balance</Table.HeaderCell>
+          <Table.HeaderCell>Rate</Table.HeaderCell>
+          <Table.HeaderCell>
+            <ButtonToOpenTokenForm
+              icon={
+                <PlusCircledIcon
+                  width={22}
+                  height={22}
+                  className="text-green9 hover:text-green11"
+                />
+              }
+            />
+          </Table.HeaderCell>
+        </Table.HeaderRow>
+        <Table.Body>
+          {initialData?.tokens?.map((token) => (
+            <TableRow token={token} key={token.symbol} />
+          ))}
+        </Table.Body>
+      </Table>
+    </div>
+  );
+}
+
+function ButtonToOpenTokenForm({
+  icon,
+  symbolToEdit,
+}: React.PropsWithChildren<{
+  icon: React.ReactElement;
+  symbolToEdit?: string;
+}>) {
+  return (
+    <Dialog
+      title="Add token parameters"
+      content={<TokenForm symbolToEdit={symbolToEdit} />}
+    >
+      <button type="button" className="leading-none">
+        {icon}
+      </button>
+    </Dialog>
+  );
+}
```

Comment:
Does this really need to be another component?

Reply:
I want to make sure that the Dialog will be the same in the edit and add buttons (just with the icon as different)